### PR TITLE
[SQL]: Allow setting a partitionkey value during document creation

### PR DESCRIPTION
**Pull request by PrashanthCorp**

### CODE REVIEW

File: src/docdb/tree/DocDBDocumentsTreeItem.ts [hunk 1]
```diff
@@ -77,4 +88,20 @@ export class DocDBDocumentsTreeItem extends DocDBTreeItemBase<RetrievedDocument>
 
         throw new UserCancelledError();
     }
+
+    // Create a nested Object given the partition key path and value
+    private createPartitionPathObject(partitionKey, partitionKeyValue): Object {
```
Comment (StephenWeatherford) [inline src/docdb/tree/DocDBDocumentsTreeItem.ts]: types for params

File: src/docdb/tree/DocDBDocumentsTreeItem.ts [hunk 2]
```diff
@@ -77,4 +88,20 @@ export class DocDBDocumentsTreeItem extends DocDBTreeItemBase<RetrievedDocument>
...
+        //remove leading slash
+        partitionKey = partitionKey.slice(1);
```
Comment (StephenWeatherford) [inline src/docdb/tree/DocDBDocumentsTreeItem.ts]: Is this guaranteed to always start with a slash?

File: src/docdb/tree/DocDBDocumentsTreeItem.ts [hunk 3]
```diff
@@ -61,9 +61,20 @@ export class DocDBDocumentsTreeItem extends DocDBTreeItemBase<RetrievedDocument>
 
         if (docID || docID === "") {
             docID = docID.trim();
+            let body = { 'id': docID };
+            const partitionKey: string | undefined = this._collection.partitionKey && this._collection.partitionKey.paths[0];
+            if (partitionKey) {
+                const partitionKeyValue: string = await vscode.window.showInputBox({
+                    prompt: `The partition key is ${partitionKey}. Enter a value for the partition key`,
```
Comment (StephenWeatherford) [inline src/docdb/tree/DocDBDocumentsTreeItem.ts]: suggest: Enter a value for the partition key (${partitionKey)}

File: src/docdb/tree/DocDBDocumentsTreeItem.ts [hunk 4]
```diff
@@ -61,9 +61,20 @@ export class DocDBDocumentsTreeItem extends DocDBTreeItemBase<RetrievedDocument>
 
         if (docID || docID === "") {
             docID = docID.trim();
+            let body = { 'id': docID };
+            const partitionKey: string | undefined = this._collection.partitionKey && this._collection.partitionKey.paths[0];
```
Comment (StephenWeatherford) [inline src/docdb/tree/DocDBDocumentsTreeItem.ts]: paths[0] is checking that the first array value has a value, but will throw if the array is empty.  Did you mean && this._collection.partitionKey.paths.length
?

File: src/docdb/tree/DocDBDocumentsTreeItem.ts (hunk 3 above)
Comment (StephenWeatherford) [inline src/docdb/tree/DocDBDocumentsTreeItem.ts]: Not sure we should assign a key value if the user enters ENTER.  What do you think?

Comment (PrashanthCorp) [reply src/docdb/tree/DocDBDocumentsTreeItem.ts]: It currently creates the nested object with an empty string. 

Looks like empty strings aren't the same as no partition key. Thanks for the catch!

File: src/docdb/tree/DocDBDocumentsTreeItem.ts [hunk 5]
```diff
@@ -65,12 +65,14 @@ export class DocDBDocumentsTreeItem extends DocDBTreeItemBase<RetrievedDocument>
...
                 const partitionKeyValue: string = await vscode.window.showInputBox({
-                    prompt: `The partition key is ${partitionKey}. Enter a value for the partition key`,
+                    prompt: `Enter a value for the partition key ("${partitionKey}")`,
                     ignoreFocusOut: true
                 });
-                // We cannot pass a partition key value during document creation.
-                // We need to present the partitionKey value as part of the document contents
-                Object.assign(body, this.createPartitionPathObject(partitionKey, partitionKeyValue));
+                if (partitionKeyValue) {
+                    // We cannot pass a partition key value during document creation.
```
Comment (StephenWeatherford) [inline src/docdb/tree/DocDBDocumentsTreeItem.ts]: Nit: comment is confusing - this *is* document creation.

File: src/docdb/tree/DocDBDocumentsTreeItem.ts [hunk 6]
```diff
@@ -90,9 +92,11 @@ export class DocDBDocumentsTreeItem extends DocDBTreeItemBase<RetrievedDocument>
     }
 
     // Create a nested Object given the partition key path and value
-    private createPartitionPathObject(partitionKey, partitionKeyValue): Object {
+    private createPartitionPathObject(partitionKey: string, partitionKeyValue: string): Object {
         //remove leading slash
-        partitionKey = partitionKey.slice(1);
+        if (partitionKey[0] === '/') {
```
Comment (StephenWeatherford) [inline src/docdb/tree/DocDBDocumentsTreeItem.ts]: You'd also need to check that it's not empty

**Pull request merged**